tests/test_path_operations.py

```diff
@@ -203,6 +203,49 @@ def test_is_file(s3_mock):
     assert S3Path('/test-bucket/build/lib/pathlib.py').is_file()
 
 
+def test_read_line(s3_mock):
+    s3 = boto3.resource('s3')
+    s3.create_bucket(Bucket='test-bucket')
+    object_summary = s3.ObjectSummary('test-bucket', 'directory/Test.test')
+    object_summary.put(Body=b'test data\ntest data')
+
+    with S3Path('/test-bucket/directory/Test.test').open("r") as fp:
+        assert fp.readline() == "test data"
```

Comment:
fp.readline() should return "test data\n" if I'm not mistaken 
See documentation [here](https://docs.python.org/3/tutorial/inputoutput.html#methods-of-file-objects)

Reply:
Yes you are right, I will fix it asap

Reply:
Hum it is not that easy, `iter_lines` from `botocore.response.StreamingBody` is removing the new line characters before yielding the line so it is complicated for us to know if the last line has a new line character or not.

We could add the newline character eachtime, but it means that `test\ntest` and `test\ntest\n` will return the same values.

Reply:
I don't like botocore :-(
for now lets keep it like this
If we will get requests to add the \n we'll think about it